### DISCUSSION TITLE
Can a venv be used as OBS's interpreter?

### DISCUSSION THREAD

**Discussion by ogtega** [upgradeQ/Streaming-Software-Scripting-Reference#32] [Q&A] [answered]
I just wrote a script that watches media player status on Linux using dbus_next.

I typically dislike installing any project dependency globally using pip, so I would rather designate a venv for obs to use for my scripts.

Is this even possible with OBS's current python script implementation?

**Comment by upgradeQ** [comment, marked as answer]
I think no. Though there is a hack, to use different version of python in GNU/Linux:
1. install `pyenv`  
2. run this `env PYTHON_CONFIGURE_OPTS="--enable-shared" pyenv install <some_python_version>` 
3. cd to .pyenv/versions/some_python_version
4. Start OBS Studio like this  `LD_PRELOAD=./lib/libpython3.11.so obs`  

Maybe you could also rewrite your program to use polling on sockets or named pipes.

**Reply by ogtega** [reply]
I just went with the file IO method, it works well enough and I only have to run my program in the background. Though a script would've been nice to implement.

Do wonder if perhaps OBS might be interested in reading the shebang at the top of a python script in the future.
